pkg/dockercfg/config.go

```diff
@@ -19,22 +23,48 @@ type DockerAuthConfigEntry struct {
 	Password string `json:"password"`
 }
 
-// ReadDockerAuthConfigFromFile attempts to read docker configJSON from a given file path.
-func ReadDockerAuthConfigFromFile(filePath string) (cfg DockerAuthConfig, err error) {
+func New() *DockerConfig {
+	return &DockerConfig{
+		Auths: map[string]DockerAuthConfigEntry{},
+	}
+}
+
+func (c *DockerConfig) AddAuth(registry, username, password string) *DockerConfig {
+	c.Auths[registry] = DockerAuthConfigEntry{Username: username, Password: password}
+	return c
+}
```

Comment:
New and AddAuth is only used by tests. I suggest we remove them, and update the tests to defined the required values directly in the object creation, for example DockerConfig{Auths:{"registry":{UserName:..., Password: ....}}}

Reply:
Fixed, also removed DockerConfig from the struct name (since its already in the package name)